### PR TITLE
chore: update code to have tsc from jsdoc

### PR DESCRIPTION
**Pull request by lpatiny**

### CODE REVIEW

**Comment by lpatiny** [comment]
@targos 

I have a problem after the update. Trying `npm run build` I get the following error:

```
Error: Invalid value "umd" for option "output.format" - UMD and IIFE output formats are not supported for code-splitting builds.
    at error (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:198:30)
    at validateOptionsForMultiChunkOutput (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:16270:16)
    at Bundle.generate (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:16104:17)
    at async /Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:23792:27
    at async catchUnfinishedHookActions (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:23209:20)
    at async runBuild (/Users/lpatiny/git/cheminfo/pubchem/node_modules/cheminfo-build/bin/cheminfo-build.js:81:3) {
  code: 'INVALID_OPTION',
  url: 'https://rollupjs.org/guide/en/#outputformat'
}
```

and I don't know what I'm doing wrong.

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Comment (targos) [inline jest.config.js:1]: Jest seems unused, so you can remove this config file and jest dependencies.

File: package.json [hunk 2]
```diff
@@ -41,36 +42,29 @@
...
-    "prettier": "^2.3.2",
-    "rollup": "^2.52.7"
+    "prettier": "^2.7.1",
+    "rimraf": "^3.0.2",
+    "typescript": "^4.7.4",
+    "vitest": "^0.22.1"
   },
   "dependencies": {
     "deepmerge": "^4.2.2",
+    "eslint-config-cheminfo-typescript": "^11.0.1",
```
Comment (targos) [inline package.json:61]: should be a devDep

File: package.json [hunk 3]
```diff
@@ -41,36 +42,29 @@
...
     "js-quantities": "^1.7.6",
     "jsonpath": "^1.1.1",
-    "ml-array-mean": "^1.1.5",
-    "ml-array-median": "^1.1.5",
-    "ml-array-standard-deviation": "^1.1.6",
-    "node-fetch": "^2.6.1",
+    "ml-array-mean": "^1.1.6",
+    "ml-array-median": "^1.1.6",
+    "ml-array-standard-deviation": "^1.1.8",
+    "node-fetch": "^3.2.10",
```
Comment (targos) [inline package.json:67]: node-fetch v3 is ESM-only

**Comment by targos** [comment]
> @targos
> 
> I have a problem after the update. Trying `npm run build` I get the following error:
> 
> ```
> Error: Invalid value "umd" for option "output.format" - UMD and IIFE output formats are not supported for code-splitting builds.
>     at error (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:198:30)
>     at validateOptionsForMultiChunkOutput (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:16270:16)
>     at Bundle.generate (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:16104:17)
>     at async /Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:23792:27
>     at async catchUnfinishedHookActions (/Users/lpatiny/git/cheminfo/pubchem/node_modules/rollup/dist/shared/rollup.js:23209:20)
>     at async runBuild (/Users/lpatiny/git/cheminfo/pubchem/node_modules/cheminfo-build/bin/cheminfo-build.js:81:3) {
>   code: 'INVALID_OPTION',
>   url: 'https://rollupjs.org/guide/en/#outputformat'
> }
> ```
> 
> and I don't know what I'm doing wrong.

I don't know, it's the first time I see this error.

File: .ncurc [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (targos) [inline .ncurc:1]: please name this file `.ncurc.json`

**Pull request merged**